plugin/session_view.py

```diff
@@ -173,20 +174,35 @@ def shutdown_async(self) -> None:
     def diagnostics_key(self, severity: int) -> str:
         return "lsp{}d{}".format(self.session.config.name, severity)
 
+    def diagnostics_tag_scope(self, tag: int) -> Optional[str]:
+        for k, v in DiagnosticTag.__dict__.items():
+            if v == tag:
+                return 'markup.{}.lsp'.format(k.lower())
+        return None
+
     def present_diagnostics_async(self, flags: int) -> None:
         data_per_severity = self.session_buffer.data_per_severity
         for severity in reversed(range(1, len(DIAGNOSTIC_SEVERITY) + 1)):
             key = self.diagnostics_key(severity)
+            key_tag = '{}_tags'.format(key)
```

Comment:
I think you need a separate key for each tag. Otherwise only one of the tag regions will show up if both tags are present and use the same severity, because the first one will just be overridden in the second loop iteration below if they both use the same key.
```python
key_tag = {tag: '{}_tags_{}'.format(key, str(tag)) for tag in data.regions_with_tag.keys()}
```
and then below
```python
self.view.add_regions(key_tag[tag], ...
```

Reply:
I guess I went a bit lazy on that part because I think it would be hard to handle the order correctly here anyway as the order might have to differ depending on what styling is used for each tag. And I can't predict what styling the user will use. But I can apply your suggestion which I guess will be slightly better anyway.

Reply:
This issue wasn't really meant to be related to diagnostics which have both tags at the same time (for a single diagnostic), but more for if there are some regions in the file with tag=1 and also some other regions in another part of the file with tag=2. Then the regions with tag=1 won't be drawn at all, because view.add_regions is called twice and iirc it doesn't actually adds the new regions to the already existing ones, but instead overrides all regions with that key with the new ones. Therefore if a file has both some "unnecessary" and some "deprecated" diagnostics, then the "unnecessary" won't be drawn at all.